internal/rootcoord/proxy_manager.go

```diff
@@ -105,13 +107,22 @@ func (p *proxyManager) startWatchEtcd(ctx context.Context, eventCh clientv3.Watc
 		case <-ctx.Done():
 			log.Warn("stop watching etcd loop")
 			return
+			// TODO @xiaocai2333: watch proxy by session WatchService.
 		case event, ok := <-eventCh:
 			if !ok {
 				log.Warn("stop watching etcd loop due to closed etcd event channel")
 				return
 			}
 			if err := event.Err(); err != nil {
-				// TODO do we need to retry watch etcd when ErrCompacted, but the init session func may not be idempotent so skip
+				if err == v3rpc.ErrCompacted {
+					err2 := p.WatchProxy()
```

Comment:
we might miss some proxy down here.

For instance:
ProxyA, ProxyB up, we called AddSessionFunc.

THen etcd err compacted, we retry and add
Found only proxyA, then we trigger AddSessionFunc for ProxyA again.
But we missed the ProxyB down.

Reply:
we must handle it in the init function.

Reply:
done